hack/generator/pkg/astmodel/storage_conversion_function.go

```diff
@@ -73,10 +76,15 @@ func NewStorageConversionFromFunction(
 		conversionDirection: ConvertFrom,
 		conversions:         make(map[string]StoragePropertyConversion),
 		knownLocals:         NewKnownLocalsSet(idFactory),
+		conversionContext:   conversionContext,
 	}
 
 	err := result.createConversions(receiver)
-	return result, err
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create '%s()'", result.name)
```

Comment:
In general the wrapping message should just provide the local context, rather than saying things like "failed" or "error" - that will be clear from the rest of the error message, and if there are multiple levels of wrapping then the "failed/error" messages get very noisy.
```suggestion
		return nil, errors.Wrapf(err, "creating '%s()'", result.name)
```